beacon/goclient/signing.go

```diff
@@ -11,6 +14,58 @@ import (
 	"github.com/pkg/errors"
 )
 
+func (gc *goClient) computeVoluntaryExitDomain(ctx context.Context) (phase0.Domain, error) {
+	specs, err := gc.client.Spec(gc.ctx, &api.SpecOpts{})
+	if err != nil {
+		return phase0.Domain{}, errors.Wrap(err, "failed to obtain specs")
+	}
+	if specs == nil {
+		return phase0.Domain{}, fmt.Errorf("failed to obtain specs")
+	}
+	if specs.Data == nil {
+		return phase0.Domain{}, fmt.Errorf("failed to obtain specs")
+	}
+
+	// TODO: consider storing fork version and genesis validators root in goClient
+	//		instead of fetching it every time
+
+	forkVersionRaw, ok := specs.Data["CAPELLA_FORK_VERSION"]
+	if !ok {
+		return phase0.Domain{}, fmt.Errorf("capella fork version not known by chain")
+	}
+	forkVersion, ok := forkVersionRaw.(phase0.Version)
+	if !ok {
+		return phase0.Domain{}, fmt.Errorf("failed to decode capella fork version")
+	}
+
+	forkData := &phase0.ForkData{
+		CurrentVersion: forkVersion,
+	}
+
+	response, err := gc.client.Genesis(ctx, &api.GenesisOpts{})
+	if err != nil {
+		return phase0.Domain{}, errors.Wrap(err, "failed to obtain genesis")
+	}
+	if response == nil {
+		return phase0.Domain{}, fmt.Errorf("failed to obtain genesis")
+	}
+	if response.Data == nil {
+		return phase0.Domain{}, fmt.Errorf("failed to obtain genesis")
+	}
+	forkData.GenesisValidatorsRoot = response.Data.GenesisValidatorsRoot
+
+	root, err := forkData.HashTreeRoot()
+	if err != nil {
+		return phase0.Domain{}, errors.Wrap(err, "failed to calculate signature domain")
+	}
+
+	var domain phase0.Domain
+	copy(domain[:], spectypes.DomainVoluntaryExit[:])
+	copy(domain[4:], root[:])
```

Comment:
nit: I don't think we'll ever change domain length but perhaps we could make its length abstract, but I don't mind leaving it as it is
```suggestion
	copy(domain[len(spectypes.DomainVoluntaryExit):], root[:])
```

Reply:
Also, I just want to confirm something:

`phase0.Domain`' length is 32, spectypes.DomainVoluntaryExit is 4 and root is 32, so last 4 bytes of `root` won't be copied, but as I understand it's ok?